openedx/core/djangoapps/user_api/accounts/views.py

```diff
@@ -581,6 +591,54 @@ def partial_update(self, request):
             return Response(text_type(exc), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+class LMSAccountRetirementView(ViewSet):
+    """
+    Provides an API endpoint for retiring a user in the LMS.
+    """
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, CanRetireUser,)
+    parser_classes = (JSONParser,)
+
+    @request_requires_username
+    def post(self, request):
+        """
+        POST /api/user/v1/accounts/retire_LMS/
```

Comment:
Yes, good thought. The first endpoint covers "mission-critical" retirement actions that need to be performed atomically. This endpoint covers tables deemed less critical and doesn't wrap it all in a transaction. I don't have any great ideas on how to differentiate them better via the endpoint names, but I am open to suggestions!

Reply:
Maybe `/retire/` and `/retire_misc/` to indicate the second endpoint is a bit of a non-critical grab-bag? They're both issued in the context of the LMS - so that suffix really reveals nothing.